packages/engine/Specs/Scene/PostProcessStageCollectionSpec.js

```diff
@@ -381,110 +381,202 @@ describe(
       expect(scene).toRender([0, 255, 255, 255]);
     });
 
-    it("uses Reinhard tonemapping", function () {
-      if (!scene.highDynamicRangeSupported) {
-        return;
+    describe("HDR tonemapping", () => {
+      const black = [0, 0, 0, 255];
+      const grey = [0.5, 0.5, 0.5, 1.0];
+      const white = [4.0, 4.0, 4.0, 1.0];
+      const red = [4.0, 0.0, 0.0, 1.0];
+      const green = [0.0, 4.0, 0.0, 1.0];
+      const blue = [0.0, 0.0, 4.0, 1.0];
+
+      /**
+       * @param {Tonemapper} tonemapper
+       * @param {number[]} inputFragColor rgba in float values to use in fragment shader
+       * @param {number[]} expectedColor rgba in rgb 0-255 to use for expected rendered colors
+       */
+      function validateTonemapper(tonemapper, inputFragColor, expectedColor) {
+        if (!scene.highDynamicRangeSupported) {
+          return;
+        }
+
+        scene.postProcessStages.tonemapper = tonemapper;
+
+        const inputColorRgb = inputFragColor.map((n) =>
+          Math.floor(Math.min(Math.max(n * 255, 0), 255))
+        );
+        const fs =
+          "void main() { \n" +
+          `    out_FragColor = vec4(${inputFragColor.join(", ")}); \n` +
+          "} \n";
+        scene.primitives.add(new ViewportPrimitive(fs));
+
+        // validate we start by rendering the expected color
+        expect(scene).withContext(`without HDR`).toRender(inputColorRgb);
+        // toggle HDR on
+        scene.highDynamicRange = true;
+        // validate we render a DIFFERENT color, not black, and that it matches the expected color
+        expect(scene).toRenderAndCall(function (rgba) {
+          expect(rgba).not.toEqual(black);
+          expect(rgba).withContext(`with HDR`).not.toEqual(inputColorRgb);
+          expect(rgba[0]).withContext("r").toEqual(expectedColor[0]);
+          expect(rgba[1]).withContext("g").toEqual(expectedColor[1]);
+          expect(rgba[2]).withContext("b").toEqual(expectedColor[2]);
+          expect(rgba[3]).withContext("a").toEqual(expectedColor[3]);
+        });
+
+        scene.highDynamicRange = false;
+        scene.primitives.removeAll();
       }
```

Comment:
I think these should be in an `afterEach` block. That way, if there is an issue with spec execution, the proper values still get restored and the rest of the tests run as expected.

Reply:
These are already included in the `afterEach` for the whole file but the first version of this helper function looped over all the colors in one spec so it was needed between loops. Forgot to take it out when I ditched that idea in favor of the separate, cleaner, independent tests I have now.